components/automate-gateway/handler/iam/v2beta/policy/policy.go

```diff
@@ -424,6 +424,25 @@ func (p *Server) ListProjects(
 	return &pb_resp.ListProjectsResp{Projects: projects}, nil
 }
 
+// IntrospectAllProjects returns the list of all projects for display in the
+// global projects filter, so it may include unassigned.
+func (p *Server) IntrospectAllProjects(
+	ctx context.Context, req *pb_req.ListProjectsReq) (*pb_resp.ListProjectsResp, error) {
+	resp, err := p.projects.ListProjects(ctx, &authz.ListProjectsReq{GlobalFilterView: true})
+	if err != nil {
+		return nil, err
+	}
+	projects := make([]*pb_common.Project, len(resp.Projects))
+	for i, proj := range resp.Projects {
+		apiProj, err := domainProjectToAPIProject(proj)
+		if err != nil {
+			return nil, err
+		}
+		projects[i] = apiProj
+	}
+	return &pb_resp.ListProjectsResp{Projects: projects}, nil
+}
+
```

Comment:
TODO: extract common code from ListProjects and IntrospectAllProjects in this file into a helper function to be DRYer.

Reply:
Roger. Point is sort of moot now anyway, with the latest commit.